share/trap.sh

```diff
@@ -63,11 +63,30 @@ trap_add()
             # __EBASH_TRAP_LEVEL is owned and updated by our trap() function. It'll update it soon.
         fi
 
-        local complete_trap
-        [[ ${sig} == "EXIT" ]] && complete_trap+="_ebash_on_exit_start; "
-        [[ -n "${cmd}"      ]] && complete_trap+="${cmd}; "
-        [[ -n "${existing}" ]] && complete_trap+="${existing}; "
```

Comment:
Now I see why you create an array to collect the commands first.  This is the line to add the extra ";" if there is multiple trap_adds.

Reply:
> Now I see why you create an array to collect the commands first. This is the line to add the extra ";" if there is multiple trap_adds.

Yes, exactly. The array is easier to deal with than a flat string. The code that was here was well-intentioned but had some minor bugs. Sometimes we'd end up with `; ; ` getting appended, which isn't valid :-|.